include/myst/thread.h

```diff
@@ -24,6 +24,7 @@
 
 #define MYST_THREAD_MAGIC 0xc79c53d9ad134ad4
 #define MYST_MAX_MUNNAP_ON_EXIT 5
+#define MYST_PID_MAX 0x8000
```

Comment:
where did this magic number come from? Is this really the case? If more than this number of processes is created will it fail?

Reply:
On the host /proc/sys/kernel/pid_max is set to 32678 (0x8000) and I thought keeping the lowest possible would be a good place to start. Currently we don't have a solution for wraping around for tid/pids once the pid_max is reached (reason why #1013 is a draft).

From man proc - 
/proc/sys/kernel/pid_max (since Linux 2.5.34)
              This file specifies the value at which PIDs wrap around
              (i.e., the value in this file is one greater than the
              maximum PID).  PIDs greater than this value are not
              allocated; thus, the value in this file also acts as a
              system-wide limit on the total number of processes and
              threads.  The default value for this file, 32768, results
              in the same range of PIDs as on earlier kernels.  On
              32-bit platforms, 32768 is the maximum value for pid_max.
              On 64-bit systems, pid_max can be set to any value up to
              2^22 (PID_MAX_LIMIT, approximately 4 million).

Reply:
ok. If this is going to be handled properly in the other PR then I will accept that. Can you put a comment in here that this is the typical default used in the host operating system.